searchlib/src/vespa/searchlib/features/dotproductfeature.cpp

```diff
@@ -367,8 +367,7 @@ SparseDotProductByCopyExecutor<A>::getAttributeValues(uint32_t docId, const AT *
     for (const IV & iv(this->_queryIndexes); (i < iv.size()) && (iv[i] < count); i++) {
         _copy[i] = _copy[iv[i]];
     }
-    values = reinterpret_cast<const AT *>(&_copy[0]);
-    return i;
+    return vespalib::ConstArrayRef(reinterpret_cast<const AT *>(&_copy[0]), i);
```

Comment:
Consider using std::vector::data() to access the raw pointer.

Reply:
Done.